src/Datadog.Trace/FrameworkDescription.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Datadog.Trace
+{
+    internal class FrameworkDescription
+    {
+        private static FrameworkDescription _instance;
+
+        private FrameworkDescription(
+            string name,
+            string productVersion,
+            string fileVersion,
+            Version assemblyVersion,
+            Version environmentVersion,
+            string osPlatform,
+            string osArchitecture,
+            string processArchitecture)
+        {
+            Name = name;
+            ProductVersion = productVersion;
+            FileVersion = fileVersion;
+            AssemblyVersion = assemblyVersion;
+            EnvironmentVersion = environmentVersion;
+            OSPlatform = osPlatform;
+            OSArchitecture = osArchitecture;
+            ProcessArchitecture = processArchitecture;
+        }
+
+        public static FrameworkDescription Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = Create();
+                }
+
+                return _instance;
+            }
+        }
+
+        public string Name { get; }
+
+        public string ProductVersion { get; }
+
+        public string FileVersion { get; }
+
+        public Version AssemblyVersion { get; }
+
+        public Version EnvironmentVersion { get; }
+
+        public string OSPlatform { get; }
+
+        public string OSArchitecture { get; }
+
+        public string ProcessArchitecture { get; }
+
+        public static string GetOSPlatform()
+        {
+            switch (Environment.OSVersion.Platform)
+            {
+                case PlatformID.Win32NT:
+                    return "win";
+                case PlatformID.Unix:
+                    return "linux";
+                default:
+                    return "unknown";
+            }
+        }
+
+        public override string ToString()
+        {
+            string architecture;
+
+            if (OSPlatform == "win" && ProcessArchitecture == "x86" && OSArchitecture == "x64")
+            {
+                architecture = "win-wow64";
+            }
+            else
+            {
+                architecture = $"{OSPlatform}-{OSArchitecture}";
+            }
+
+            return $"{Name} {ProductVersion} {architecture}";
+        }
+
+        private static FrameworkDescription Create()
+        {
+            var assembly = typeof(object).GetTypeInfo().Assembly;
+            var assemblyName = assembly.GetName();
+
+            var fileVersionAttribute = (AssemblyFileVersionAttribute)assembly.GetCustomAttribute(typeof(AssemblyFileVersionAttribute));
+            var fileVersion = fileVersionAttribute.Version;
+
+            var osArchitecture = Environment.Is64BitOperatingSystem
```

Comment:
.NET Core is supported on ARM32 and ARM64 for several platforms, so we need to make sure we can count those, in addition to `x64` and `x86`.

Reply:
I was trying to remove the dependency on `RuntimeInformation`, but I put that back in for now. I was focusing on automatic instrumentation (which only supports win-x86, win-x64, and linux-x64) and forgot about manual instrumentation which should be able to run on any platform that supports .NET Standard 2.0.